pysat/_files.py

```diff
@@ -5,64 +5,93 @@
 # ----------------------------------------------------------------------------
 
 import datetime as dt
+from functools import partial
 import numpy as np
 import os
 import warnings
-import weakref
 
 import pandas as pds
-from pysat import data_dir as data_dir
+import pysat  # Needed to access pysat.params across reimports
+from pysat import logger, Instrument
```

Comment:
Since we're importing pysat, recommend using `logger = pysat.logger` and `pysat.Instrument` below and removing second line.